tiatoolbox/annotation/storage.py

```diff
@@ -1007,6 +1050,195 @@ def select_values(
             select, unique, squeeze, items, select_values
         )
 
+    def nquery(
+        self,
+        geometry: Optional[Geometry] = None,
+        where: Optional[Predicate] = None,
+        n_where: Optional[Predicate] = None,
+        distance: float = 5.0,
+        geometry_predicate: str = "intersects",
+        mode: str = "poly-poly",
+    ) -> Dict[str, Dict[str, Annotation]]:
+        """Query for annotations within a distance of another annotation.
+
+        Args:
+            geometry (Geometry):
+                A geometry to use for the query. If None, the geometry of
+                the annotation is used. Defaults to None.
```

Comment:
This isnt clear. I'd suggest something like:
A geometry to query for annotations within (according to geometry_predicate). If None, all annotations in the store are considered. Defaults to None.

Reply:
@measty Is this better now? See the latest commit.

Reply:
Yes, looks fine now